### ISSUE TITLE
Button: icon-only buttons violate accessibility standards

### ISSUE THREAD

**Issue by wkeese** [ibm-js/deliteful#600]
Apparently all buttons need to have text (not a `title` or `alt` or `aria-label`, but actual inline text).   If you want the button to just appear as an icon, then the text should be hidden, but it still needs to be there.

This is documented in https://www.nczonline.net/blog/2013/04/01/making-accessible-icon-buttons/ , https://www.joedolson.com/2014/06/visual-icon-problem/, and https://www.filamentgroup.com/lab/bulletproof_icon_fonts.html.

To solve this problem, perhaps if the button's label is undefined, then it should insert the `title` (or `aria-label`?) text in its place, but make it hidden?

**Comment by wkeese** [comment]
Hmm, I don't there's any real issue.  With the current code, both JAWS (on Windows) and VoiceOver (on iPad) will read the `title` attribute of the button.  And sighted users can see the button's title by mousing over it.

**Comment by wkeese** [comment]
Note that the following template works around the spurious error message:

```xml
<template attach-point="focusNode">
  	<span attach-point="iconNode" class="d-icon {{this.iconClass ? this.iconClass : 'd-hidden'}}"
  		aria-hidden="true"></span
	><span attach-point="labelNode" d-hidden="{{!this.label}}"
		>{{this.label || this.getAttribute("aria-label") || this.title}}</span>
</template> 
```
